### PR TITLE
CLOUDP 167794 : Migrate "Cloud Provider Access" CLI stores to v2 sdk

### PR DESCRIPTION
**Pull request by swattyT**
## Proposed changes

_Jira ticket:_ CLOUDP-# [CLOUDP-167794](https://jira.mongodb.org/browse/CLOUDP-167794)
EVG patch:
https://spruce.mongodb.com/version/6425b079850e61931dd03efa/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Closes #[issue number]

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by wtrocki** [comment]
Looks very good. Did you tried atlas setup command?

**Comment by wtrocki** [comment]
Did not approve as I'm not sure if we want to merge more commands to v2client.

File: internal/store/access_role.go
Comment (wtrocki) [inline internal/store/access_role.go:28]: nit - we might decide to use v2 as shortcut to make functions shorter. Just though

**Comment by matt-condon** [comment]
> Did you tried running atlas setup command

@swattyT happy to merge this one provided this is passing.

I assume e2e tests are ok too?

**Comment by swattyT** [comment]
> > Did you tried running atlas setup command
> 
> @swattyT happy to merge this one provided this has been validated.
> 
> I assume e2e tests are ok too?

I ran the create and list commands locally and [atlas_generic_e2e](https://spruce.mongodb.com/task/mongocli_master_e2e_generic_atlas_generic_e2e_patch_81f3ba16bc3d5e7d3430839ee186be1b2da32803_641c60df562343770e708c35_23_03_23_14_23_37?execution=0) have tests for AccessRole.

**Comment by wtrocki** [comment]
We need to test setup as well as it contains those commands and it is not covered by e2e

**Review by matt-condon — APPROVED** [review]

File: internal/cli/atlas/cloudproviders/accessroles/aws/create.go
Comment (sarahsimpers) [inline internal/cli/atlas/cloudproviders/accessroles/aws/create.go:32]: Will the CLI actually start displaying a different value, like `Rolename.RoleID`? If not (and this is just a new variable for the same RoleID), this change causes the automation we put in place for outputs to not work correctly. You would need to remove lines 68-70 of this file.

Comment (wtrocki) [reply internal/cli/atlas/cloudproviders/accessroles/aws/create.go:32]: @sarahsimpers it will be the same roleId just new variable name. I noticed that docs use unresolved templates in documentation (so this will be affected) but actual value never changes.

**Review by wtrocki — APPROVED** [review]

File: internal/cli/atlas/cloudproviders/accessroles/aws/create.go
Comment (swattyT) [reply internal/cli/atlas/cloudproviders/accessroles/aws/create.go:32]: Updated the template in the Annotations so the docs generated remain the same.
I did not create a separate variable just to avoid confusion with the output template.

**Review by matt-condon — APPROVED** [review]
LGTM

**Pull request merged**
